src/profile-logic/call-node-info.js

```diff
@@ -331,21 +783,528 @@ export class CallNodeInfoInvertedImpl
+    return [rangeStart, rangeEnd];
+  }
+
+  /**
+   * Materialize inverted call nodes for parentNodeHandle's children in the
+   * inverted tree.
+   *
+   * The returned array of call node handles is sorted by func.
+   */
+  _createChildren(
+    parentNodeHandle: InvertedCallNodeHandle
+  ): InvertedCallNodeHandle[] {
+    const parentDeepNodes =
+      this._takeDeepNodesForInvertedNode(parentNodeHandle);
+    const childrenInfo = this._computeChildrenInfo(
+      parentNodeHandle,
+      parentDeepNodes
     );
+    if (childrenInfo === null) {
+      // This node has no children.
+      return [];
+    }
+
+    return this._createChildrenForInfo(childrenInfo, parentNodeHandle);
+  }
+
+  /**
+   * Compute the information needed to create the children of parentNodeHandle.
+   *
+   * As we go deeper into the inverted tree, we go higher up in the non-inverted
+   * tree: To create the children of an inverted node, we need to look at the
+   * parents / "prefixes" of the corresponding non-inverted "deep nodes".
+   *
+   * See the class documentation for more details and examples.
+   */
+  _computeChildrenInfo(
+    parentNodeHandle: InvertedCallNodeHandle,
+    parentDeepNodes: Uint32Array
+  ): ChildrenInfo | null {
+    const parentDeepNodeCount = parentDeepNodes.length;
+    const [parentIndexRangeStart, parentIndexRangeEnd] =
+      this.getSuffixOrderIndexRangeForCallNode(parentNodeHandle);
+    if (parentIndexRangeStart + parentDeepNodeCount !== parentIndexRangeEnd) {
+      throw new Error('indexes out of sync');
+    }
+
+    const callNodeTable = this._callNodeTable;
+
+    // Count how many of the parent's deep nodes end at the parent. If there
+    // are any, they will all be at the start of the parentDeepNodes array by
+    // construction of the suffix order.
+    let nodesWhichEndHereCount = 0;
+    while (nodesWhichEndHereCount < parentDeepNodeCount) {
+      const deepNode = parentDeepNodes[nodesWhichEndHereCount];
+      if (callNodeTable.prefix[deepNode] !== -1) {
+        break;
+      }
+      nodesWhichEndHereCount++;
+    }
+
+    if (nodesWhichEndHereCount === parentDeepNodeCount) {
+      // All deep nodes ended at the parent's depth. The parent has no children.
+      return null;
+    }
+
+    const childrenDeepNodeCount = parentDeepNodeCount - nodesWhichEndHereCount;
+    const childrenDeepNodes = new Uint32Array(childrenDeepNodeCount);
+    // assert(childrenDeepNodeCount > 0);
+
+    // Iterate over the remaining deep nodes, get each deep node's prefix,
+    // and build up our list of children. For each child, compute its func and
+    // its number of deep nodes.
+
+    const firstChildFirstParentDeepNode =
+      parentDeepNodes[nodesWhichEndHereCount];
+    const firstChildFirstDeepNode =
+      callNodeTable.prefix[firstChildFirstParentDeepNode];
+    childrenDeepNodes[0] = firstChildFirstDeepNode;
+    const firstChildFunc = callNodeTable.func[firstChildFirstDeepNode];
```

Comment:
Without looking too close, I wonder why this can't be part of the loop below?

(but I see this code is removed later anyway so let's skip this comment)